custodialwalletutils/README.md

```diff
@@ -22,58 +22,37 @@
-
-  The tests are written using a testing framework, which provides a set of tools and functions for defining and running tests. The tests themselves are written in JavaScript and are designed to be run in the Node.js environment. When a test is run, it will call the function or class being tested with a set of predefined inputs and then check that the output is as expected. If the output matches the expected result, the test passes. If not, the test fails, indicating that there is a problem with the code.
+The `custodialwalletutils` library implements a factory-strategy pattern, enabling object creation without specifying their exact class. The library's structure is divided into two primary directories:
 
-  The `tests/` directory is an essential part of the development process. Before any changes are made to the codebase, the tests should be run to ensure that the existing code is working correctly. After changes are made, the tests should be run again to ensure that the changes have not introduced any new bugs. If new functionality is added to the codebase, new tests should be written to test this functionality.
+1. **src/**: This directory houses the core source code and is organized into subdirectories:
+  - **factories/**: Contains logic to return the correct strategy based on received configuration.
+  - **models/**: Hosts models for interacting with the library's exposed services, including the `SignatureRequest`.
+  - **services/**: Includes `CustodialWalletService`, which exposes the library's services.
+  - **strategies/**: Comprises configuration classes for the factory and individual strategy classes, each with specific strategy logic.
+  - **utils/**: Stores reusable logic utilized across different strategies.
 
-In addition to ensuring the correctness of the code, the tests also serve as a form of documentation. They provide examples of how the functions and classes should be used and what their expected behavior is. This can be very helpful for developers who are new to the codebase or who are working on complex features.
-
-The custodialwalletutils library is designed to be easy to use, with a focus on developer experience and code readability. It provides a robust suite of utility functions and classes, making it a dependable choice for managing custodial wallets in JavaScript applications.
+2. **tests/**: Contains all unit tests for the utility functions and classes in the library. Each test file corresponds to a source file and includes tests for various functions and classes.
 
+The strategy classes in the src/ directory, following a factory-strategy pattern, along with various helper functions and classes, facilitate the creation and management of custodial wallets, enhancing code readability and maintainability.
 ## 3. Content
 
```

Comment:
Was a bit lost on what to write here, saved it for last, but everything fell into place...